### PR TITLE
Build a matrix of java/mvn builders.

### PR DESCRIPTION
**Pull request by skelterjohn**
Build a matrix of java/mvn builders, crossing maven 3.5.0/3.3.9 with java 7/8.

Use build args to parameterize the base image. Copypaste in cloudbuild.yaml. Possibly generate it with a script in the future.

### CODE REVIEW

**Comment by bendory** [comment]
LGTM

File: java/mvn/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM openjdk:8-jdk-alpine
+ARG BASE_IMAGE
```
Comment (skelterjohn) [inline java/mvn/Dockerfile]: This needs a default value.

File: java/mvn/cloudbuild.yaml [hunk 2]
```diff
@@ -2,8 +2,45 @@
...
+  - '--build-arg=BASE_IMAGE=openjdk:7-jdk-alpine'
+  - '--build-arg=MAVEN_VERSION=3.5.0'
+  - '--build-arg=SHA=beb91419245395bd69a4a6edad5ca3ec1a8b64e41457672dc687c173a495f034'
+  - '--tag=gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-7'
+  - '.'
+  wait_for: ['-']
+
+# Tag a default builder version.
 - name: 'gcr.io/cloud-builders/docker'
   args: ['tag', 'gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-8', 'gcr.io/$PROJECT_ID/java/mvn']
```
Comment (aslo) [inline java/mvn/cloudbuild.yaml]: Does it make sense to run the examples with more of these new images? Or at least with the :latest image?

File: java/mvn/cloudbuild.yaml [hunk 3]
```diff
@@ -2,8 +2,45 @@
...
```
Comment (aslo) [inline java/mvn/cloudbuild.yaml]: This step only needs to `waitFor` the step that builds the `3.5.0-jdk-8` image, right?

Comment (skelterjohn) [inline java/mvn/cloudbuild.yaml]: Technically that is true, but it takes no time so I didn't bother making it run concurrently.

File: java/mvn/cloudbuild.yaml (hunk 2 above)
Comment (skelterjohn) [inline java/mvn/cloudbuild.yaml]: done

Comment (aslo) [inline java/mvn/cloudbuild.yaml]: Thoughts on parallelizing running each of the examples?

Comment (skelterjohn) [inline java/mvn/cloudbuild.yaml]: Since they share /workspace I was worried about them clobbering each other in flight. Is that not a concern?

Comment (aslo) [inline java/mvn/cloudbuild.yaml]: Ah, right. Good point.

**Pull request merged**